### ISSUE TITLE
Avoid confirmation for non-destructive actions

### ISSUE THREAD

**Issue by polytypic** [ocaml/opam#5463]
tl;dr opam frequently prompts for confirmation even for non-destructive actions. It would be nice if one could adjust the threshold for prompting (beyond just `-y`) so that opam would only prompt in cases where there is a high likelyhood of something destructive happening.

opam prompts for confirmation in various cases. For example, I just ran `opam install current` and got the following output:

```sh
➜  ocaml opam install current
The following actions will be performed:
  - install   conf-sqlite3      1      [required by sqlite3]
  - install   conf-libev        4-12   [required by current]
  - install   asetmap           0.8.1  [required by prometheus]
  - install   current_incr      0.6.1  [required by current]
  - install   sqlite3           5.1.0  [required by current]
  - recompile lwt               5.6.1  [uses conf-libev]
  - install   prometheus        1.2    [required by current]
  - recompile lwt_ssl           1.2.0  [uses lwt]
  - recompile lwt_react         1.2.0  [uses lwt]
  - recompile logs              0.7.0  [uses lwt]
  - recompile js_of_ocaml-lwt   5.0.1  [uses lwt]
  - recompile mirage-crypto-rng 0.11.0 [uses logs]
  - recompile mdx               2.2.1  [uses logs]
  - recompile lambda-term       3.3.1  [uses lwt]
  - recompile conduit           6.2.0  [uses logs]
  - recompile cohttp-lwt        5.0.0  [uses lwt]
  - recompile bos               0.2.1  [uses logs]
  - recompile mirage-crypto-pk  0.11.0 [uses mirage-crypto-rng]
  - recompile mirage-crypto-ec  0.11.0 [uses mirage-crypto-rng]
  - recompile utop              2.11.0 [uses lwt]
  - recompile conduit-lwt       6.2.0  [uses lwt]
  - recompile github            4.4.1  [uses lwt]
  - recompile cohttp-lwt-jsoo   5.0.0  [uses lwt]
  - install   current           0.6.4
  - recompile dune-release      1.6.2  [uses logs]
  - recompile x509              0.16.4 [uses logs]
  - recompile ca-certs          0.2.3  [uses logs]
  - recompile conduit-lwt-unix  6.2.0  [uses lwt]
  - recompile cohttp-lwt-unix   5.0.0  [uses lwt]
  - recompile github-unix       4.4.1  [uses lwt]
  - recompile opam-publish      2.2.0  [uses lwt_ssl]
===== 7 to install | 24 to recompile =====
Do you want to continue? [Y/n]
```

If you go through the (long) list of actions, you will notice that there is nothing "destructive" there. A number of required dependencies need to be installed and then a number of packages need to be recompiled. I generally would prefer not to be prompted for confirmation for non-destructive actions such as recompiles and installation of dependencies.

On the other hand, sometimes the confirmations save me from having to spend time reverting the actions. (Here is a contrived example:) For example, running `opam install mtime=1.4.0` I get:

```sh
➜  ocaml opam install mtime=1.4.0
The following actions will be performed:
  - downgrade mtime 2.0.0 to 1.4.0
  - downgrade eio   0.8.1 to 0.7   [uses mtime]
===== 2 to downgrade =====
Do you want to continue? [Y/n]
```

As a result of installing, I'd need to downgrade other packages. Sometimes that happens (due to transitive dependencies) even without explicitly requesting a specific version as above. For destructive actions like downgrading packages asking for a confirmation makes sense.

In other words, I wouldn't want to just blindly add `-y`. I'd rather have more fine grained way to avoid confirming non-destructive actions.

**Comment by kit-ty-kate** [comment]
As discussed previously I don't think this should be the new default but this would definitely be a good addition to the `OPAMCONFIRMLEVEL` / `--confirm-level` option.

Does anyone have a suggestion on the name of the new value? Current accepted values are: `ask`, `no`, `yes` and `unsafe-yes`. Maybe something like `non-destructive-ask` ?

**Comment by samoht** [comment]
Maybe auto? (To match the other three-state options we already have)
